### ISSUE TITLE
[BUG] Getting "error-invalid-customfield-json" on signup with Custom Fields

### ISSUE THREAD

**Issue by rafaelks** [RocketChat/Rocket.Chat#8448]
### Description:

We're working to add support to Custom Fields into the native iOS app and in my tests, I can't have Custom Fields and create a new user from the web.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.58.2
* Operating System: Linux
* Deployment Method(snap/docker/tar/etc): hosting
* Number of Running Instances: 1
* DB Replicaset Oplog: Enabled
* Node Version: v4.8.4

### Steps to Reproduce:

1. Login with administrator user;
2. Go to Administration -> Accounts -> Registration (tap Expand);
3. Copy the example JSON for Custom Fields;
4. Paste it on the Custom Fields text view;
5. Save the changes;
6. Logout from the user;
7. On the login screen, try to create a new user;

![screen shot 2017-10-10 at 8 47 44 am](https://user-images.githubusercontent.com/551004/31385151-750017ee-ad98-11e7-9607-be0e0af06927.png)

### Expected behavior:

1. User should be prompt of custom fields on the registration field;
2. **OR** user should be prompt of custom fields after registering the user;

### Actual behavior:

1. User can't create any new accounts into the server;

**Comment by rafaelks** [comment]
Looks like the error happens because there's an invalid character. I did fix this issue in our docs (https://github.com/RocketChat/Rocket.Chat.Docs/pull/410) but I don't know exactly how to fix in the Admin docs, since that JSON is being used as a [translated string](https://github.com/RocketChat/Rocket.Chat/blob/fb880acb11f9b59d89f6a2823729d26e97f50730/packages/rocketchat-i18n/i18n/en.i18n.json#L40) (?).

**Comment by maxpen** [comment]
I get this issue also on my default rocketchat instance in browser. Not just for IOS APP.

**Comment by rafaelks** [comment]
@maxpen Looks like the JSON is invalid when this error happens. Make sure you use some JSON linter to validate it and remove any invalid character: http://jsonlint.com.

After that, everything should work just fine.

**Comment by maxpen** [comment]
@rafaelks Can you copy me a working example? I'm not so good with json and that tool just tells me SyntaxError: Bad string on line 1 and I have no idea what is wrong with line 1 :)

**Comment by maxpen** [comment]
@MartinSchoeler that works! When do we see this pr in the new release?

**Comment by jszaszvari** [comment]
I'm unable to get any customfields to show up in the UserInfo tab as well. I can see it when i go into My Account just not when I look at a users profile.

**Comment by si90** [comment]
The problem is really annoying, it would be great if you could change the behavior like described from 

> rafaelks

**Comment by rafaelks** [comment]
Tried again this today... took me many tries until I got it right. There's no kind of JSON validation on the field and copying it from the documentation doesn't work because of hidden characters.

In case anyone wants to try the documentation JSON, here's one working:

```json
{
    "role": {
        "type": "select",
        "defaultValue": "student",
        "options": ["teacher", "student"],
        "required": true,
        "modifyRecordField": {
            "array": true,
            "field": "roles"
        }
    },
    "twitter": {
        "type": "text",
        "required": true,
        "minLength": 2,
        "maxLength": 10
    }
}
```

**Comment by vynmera** [comment]
yeah, the documentation below it is garbage, it uses tabs which doesnt work

**Comment by arminfelder** [comment]
I experienced the same issue, I guess we should let RocketChat sanetize the JSON , by removing tabs

**Comment by robincafolla** [comment]
I don't believe this issue should have been closed? Certainly it's still present in v3.14.0
